clearnode/channel_service.go

```diff
@@ -103,76 +103,110 @@ func (s *ChannelService) RequestCreate(wallet common.Address, params *CreateChan
 }
 
 func (s *ChannelService) RequestResize(params *ResizeChannelParams, rpcSigners map[string]struct{}, logger Logger) (ChannelOperationResponse, error) {
-	channel, err := GetChannelByID(s.db, params.ChannelID)
-	if err != nil {
-		logger.Error("failed to find channel", "error", err)
-		return ChannelOperationResponse{}, RPCErrorf("failed to find channel: %s", params.ChannelID)
-	}
-	if channel == nil {
-		return ChannelOperationResponse{}, RPCErrorf("channel %s not found", params.ChannelID)
-	}
+	var channel *Channel
+	var allocations []nitrolite.Allocation
+	err := s.db.Transaction(func(tx *gorm.DB) error {
+		var err error
+		channel, err = GetChannelByID(tx, params.ChannelID)
+		if err != nil {
+			logger.Error("failed to find channel", "error", err)
+			return RPCErrorf("failed to find channel: %s", params.ChannelID)
+		}
+		if channel == nil {
+			return RPCErrorf("channel %s not found", params.ChannelID)
+		}
```

Comment:
Is there a logical difference between when an error or nil channel is returned?
Is it possible NOT to return nil channel?